examples/circuit_random/circuit_random.cc

```diff
@@ -0,0 +1,198 @@
+#include <gcs/problem.hh>
+#include <gcs/solve.hh>
+
+#include <boost/program_options.hpp>
+#include <gcs/constraints/circuit/circuit.hh>
+#include <gcs/constraints/comparison.hh>
+#include <gcs/constraints/equals.hh>
+#include <gcs/constraints/in.hh>
+#include <gcs/constraints/not_equals.hh>
+#include <iostream>
+#include <random>
+
+using namespace gcs;
+
+using std::cerr;
+using std::cmp_less;
+using std::cout;
+using std::endl;
+using std::make_optional;
+using std::make_pair;
+using std::mt19937;
+using std::nullopt;
+using std::pair;
+using std::random_device;
+using std::string;
+using std::uniform_real_distribution;
+using std::vector;
+namespace po = boost::program_options;
+
+static const double EDGE_PROBABILITY = 0.7;
+
+auto create_graph_from_seed(int n, double p, unsigned int seed) -> pair<vector<vector<long>>, unsigned int>
+{
+    std::mt19937 gen(seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+
+    std::vector<std::vector<long>> distances(n);
+
+    for (int i = 0; i < n; ++i) {
+        for (int j = 0; j < n; ++j) {
+
+            if (dis(gen) < p && i != j) {
+                distances[i].push_back(int(dis(gen) * 100)); // Edge from i to j
+            }
+            else {
+                distances[i].push_back(-1);
+            }
+        }
+    }
+
+    return make_pair(distances, seed);
+}
+
+auto generate_random_graph(int n, double p) -> pair<vector<vector<long>>, unsigned int>
+{
+    std::random_device rd;
+    auto seed = rd();
+    return create_graph_from_seed(n, p, seed);
+}
+
+Stats run_circuit_problem(int n, const vector<vector<long>> & distances)
+{
+    Problem p;
+    auto x = p.create_integer_variable_vector(n, 0_i, Integer{n - 1});
+    //            for (int i = 0; i < n; ++i) {
+    //                for (int j = 0; cmp_less(j, distances[i].size()); ++j) {
+    //                    cout << distances[i][j] << " ";
+    //                }
+    //                cout << endl;
+    //            }
+
+    for (int loc1 = 0; loc1 < n; loc1++) {
+        for (int loc2 = 0; loc2 < n; loc2++) {
+            if (distances[loc1][loc2] < 0) {
+                p.post(NotEquals{x[loc1], ConstantIntegerVariableID{Integer{loc2}}});
+            }
+        }
+    }
+    SCCOptions options = {false, true, false, false, false, true};
```

Comment:
I wonder whether it would be helpful to have these as command line options, so we can run a bunch of configurations automatically as part of the tests?